src/lib/runner.rs

```diff
@@ -122,18 +122,37 @@ pub fn generate(missile: &Missile, launch_parameters: &LaunchParameter, timestep
 
 
 		let engine_stage;
-
-		if (f64::from(i) * timestep) < missile.timefire0 {
-			a = ((missile.force0 - drag_force) / missile.mass) - gravity;
-			engine_stage = "0";
-		} else if missile.timefire1 != 0.0 && (f64::from(i) * timestep) < missile.timefire1 {
-			a = ((missile.force1 - drag_force) / missile.mass) - gravity;
-			engine_stage = "1";
-		} else {
-			a = (-drag_force / missile.mass_end) - gravity;
-			engine_stage = "-";
+		let mass;
+		let force;
+
+		let burn_0 = 0.0..missile.timefire0;
+		let burn_1 = burn_0.end..missile.timefire1;
+
+		let burn_time = f64::from(i) * timestep;
+		let compute_mass = |mass_end, timefire| {
```

Comment:
For stage 2  (engine_stage = "1") this needs to be calculated from mass_end and mass_end1, another parameter is needed

Reply:
Fixed